protos/file.proto

```diff
@@ -78,13 +292,49 @@ message Metadata {
     //   position = stats_page_table[5][0];
     //   length = stats_page_table[5][1];
     // ```
-      uint64 page_table_position = 3;
+    uint64 page_table_position = 3;
   }
 
+  // File statistics.  This field is only present in Lance version 1.
+  //
+  // In Lance version 2 statistics are stored per-column and this field
+  // will not be present.
   StatisticsMetadata statistics = 5;
+
+  // The number of columns in the file (including inner columns when there
+  // are nested fields)
+  //
+  // This can be used to access the column metadata offsets array which is
+  // stored immediately before manifest_position.
+  //
+  // Given N columns the column metadata positions and sizes are stored in a
+  // contiguous buffer of 2*N uint64 values immediately preceding the file
+  // descriptor (or the metadata if the file is not self describing).
+  //
+  // If we let `column_offsets_pos` be:
+  //   manifest_position - (16 * num_columns)
```

Comment:
I wonder if we should just have a direct offset here. That would make it possible to have another optional section later.

Reply:
I'm not sure I understand.

Reply:
As in `column_offsets_pos` should be an actual field, and not just an implicit computed one. Then we could add a new section between the column metadata and the file metadata.

Reply:
Ok, I get it now.  I'll add that in.